datumaro/plugins/data_formats/datumaro/exporter.py

```diff
@@ -84,51 +90,47 @@ def context_save_media(self, item: DatasetItem, encryption: bool = False) -> Non
             image = item.media_as(Image)
             path = image.path
 
-            if self._context._save_media:
+            if context.save_media:
                 # Temporarily update image path and save it.
                 image._path = osp.join(
-                    self._context._images_dir, item.subset, self._context._make_image_filename(item)
-                )
-                self._context._save_image(
-                    item, image.path, crypter=self._crypter if encryption else NULL_CRYPTER
+                    context.images_dir, item.subset, context.make_image_filename(item)
                 )
+                context.save_image(item, encryption=encryption, path=image.path)
 
             yield
             image._path = path
         elif isinstance(item.media, PointCloud):
             pcd = item.media_as(PointCloud)
             path = pcd.path
 
-            if self._context._save_media:
+            if context.save_media:
                 # Temporarily update pcd path and save it.
-                pcd._path = osp.join(
-                    self._context._pcd_dir, item.subset, self._context._make_pcd_filename(item)
-                )
-                self._context._save_point_cloud(item, pcd.path)
+                pcd._path = osp.join(context.pcd_dir, item.subset, context.make_pcd_filename(item))
+                context.save_point_cloud(item, pcd.path)
 
                 # Temporarily update pcd related images paths and save them.
                 for i, img in enumerate(sorted(pcd.extra_images, key=lambda v: v.path)):
                     img.__path = img.path
                     img._path = osp.join(
-                        self._context._related_images_dir,
+                        context.related_images_dir,
                         item.subset,
                         item.id,
-                        f"image_{i}{self._context._find_image_ext(img)}",
+                        f"image_{i}{context.find_image_ext(img)}",
                     )
 
                     if img.has_data:
                         img.save(img.path)
 
             yield
             pcd._path = path
-            if self._context._save_media:
+            if context.save_media:
                 for img in pcd.extra_images:
                     img._path = img.__path
                     del img.__path
         else:
             raise NotImplementedError
 
-    def add_item(self, item: DatasetItem):
+    def add_item(self, item: DatasetItem, pool: Optional[Pool] = None):
```

Comment:
How about just accepting all arguments since `_SubsetWriter` could parents of other than one from `datumaro_binary`?
```suggestion
    def add_item(self, item: DatasetItem, *args, **kwargs):
```

Reply:
I added `*args` and `**kwargs` to the function parameters.